src/main/java/org/springframework/data/couchbase/repository/query/N1qlQueryCreator.java

```diff
@@ -187,17 +187,28 @@ private QueryCriteria from(final Part part, final CouchbasePersistentProperty pr
 		}
 	}
 
+	/**
+	 * Translate meta-fields to META(bucketName).id, cas, expiry.<br>
+	 * If bucketName is null, META().id etc, <br>
+	 * If not a meta-field, just create the corresponding path
+	 *
+	 * @param bucketName
+	 * @param persistentPropertyPath
+	 * @param property
+	 * @param entity
+	 * @return N1QLExpression
+	 */
 	public static N1QLExpression addMetaIfRequired(String bucketName,
 			final PersistentPropertyPath<CouchbasePersistentProperty> persistentPropertyPath,
 			final CouchbasePersistentProperty property, final PersistentEntity entity) {
 		if (entity != null && property == entity.getIdProperty()) {
-			return path(meta(i(bucketName)), i(META_ID_PROPERTY));
+			return path(meta(bucketName != null ? i(bucketName) : x("")), i(META_ID_PROPERTY));
```

Comment:
can the bucket name be empty?

Reply:
Yes. That's actually what the fix is (line 79 above).  When creating the N1qlQuery (a query derived from a method name), the collection name is not yet available so it just uses meta().id instead of meta(collectionName).id. - which is fine, because in derived queries, meta().id is unambiguous - the collection/bucket is determined by the FROM clause which is constructed later when the collection name is available.